src/H5Spoint.c

```diff
@@ -800,12 +800,15 @@ H5S__copy_pnt_list(const H5S_pnt_list_t *src, unsigned rank)
 
     /* Sanity checks */
     assert(src);
-    assert(rank > 0);
 
     /* Allocate room for the head of the point list */
     if (NULL == (dst = H5FL_CALLOC(H5S_pnt_list_t)))
         HGOTO_ERROR(H5E_DATASPACE, H5E_CANTALLOC, NULL, "can't allocate point list node");
 
+    /* If source dataspace has no extent, just return new point list node */
+    if (rank == 0)
+        HGOTO_DONE(dst);
```

Comment:
@qkoziol I'm considering revising this. While it makes sense to me to return an empty point list node when the dataspace's extent has 0 dimensions (as we can't even tell how to copy the points over in that case), this causes problems elsewhere where the library expects `space->select.sel_info.pnt_lst->head` to be non-NULL. I have a fix for that as well, but I'm thinking of just having `H5Scopy()` / `H5Sselect_copy()` return failure for point and hyperslab selection dataspaces with a 0 dimension extent, since they both don't support `H5S_SCALAR` or `H5S_NULL` dataspaces and can only end up here when `H5Sset_extent_none()` is called on them. Thoughts?

Reply:
That seems like a better solution to me. I suppose it makes sense to reset the selection to 'all' since that's what dataspaces start with?

Reply:
Agree, back to 'all' for all changes in rank would be consistent with dataspace creation.